Docs scaffold (#75)
mmore500/downstream: docs/docs/contributing.md

### Language Implementations

Each language implementation exists on its own corresponding branch (e.g., Python on the `python` branch, C++ on the `cpp` branch) with the following structure:

- Core algorithm variants under `dstream/` directory including:
      - `steady_algo`
      - `stretched_algo`
      - `tilted_algo`

>       - Optional hybrid variants (e.g., `hybrid_0_steady_1_stretched_2_algo`)

Reviewer: ```suggestion
      - as needed for particular use cases, secondary algorithms and hybrid variants (e.g., `hybrid_0_steady_1_stretched_2_algo`)
```

Reviewer: Should also mention that each algorithm comprises coresponding site selection and lookup implementations

> - Auxiliary functions in `_auxlib` module

Reviewer: ```suggestion

Internal support code is organized into an accompanying `_auxlib` module.
```

> If you create an implementation in another language, we're happy to either link to your repository or include it directly in ours on a dedicated branch. New implementations should follow the existing structure and include the CLI-based tester/debugger for compatibility with our testing framework (see the Python CI workflow for examples of how tests are run across implementations).

Reviewer: ```suggestion
Additional implementations and outside contributions are welcome!
If you create an implementation in another language, we're happy to either link to your repository or host it directly in ours on a dedicated branch.
New implementations should follow the existing conventions and structure, as far as possible.
Also consider implementing the project's established CLI interface for testing/debugging, which will enable out-of-the-box compatibility with our testing framework --- see the cpp or zig CI workflow for examples of how tests are run across implementations.
```

> If you'd like to request that we implement a specific language, please open an issue.

Reviewer: ```suggestion
If you'd like to request support for an additional programming language, please open an issue!
```

> All contributions must follow our code of conduct. This Code of Conduct is adapted from the Contributor Covenant, version 2.0, available at [https://www.contributor-covenant.org/version/2/0/code_of_conduct.html](https://www.contributor-covenant.org/version/2/0/code_of_conduct.html).

Reviewer: ```suggestion
Contributions to the repository are governed by our Code of Conduct, based on the [Contributor Covenant, version 2.0](https://www.contributor-covenant.org/version/2/0/code_of_conduct.html).
```
